sotodlib/tod_ops/sub_polyf.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np 
+
+from sotodlib import flags
+
+
+
+def get_leftright_array(
+    tod, az=None
+):
+    if az is None:
+        az = tod.boresight.az
+    
+    daz = np.copy(np.diff(tod.boresight.az))
+    daz = np.append(daz,daz[-1])
+    signs = np.sign(daz)
+    
+    # positive range
+    positive_indices = np.where(signs > 0)[0]
+    ranges = np.split(positive_indices, np.where(np.diff(positive_indices) != 1)[0]+1)
+    index_ranges_p = [(rng[0], rng[-1]) for rng in ranges]
+    
+    # negative range
+    negative_indices = np.where(signs <= 0)[0]
+    ranges = np.split(negative_indices, np.where(np.diff(negative_indices) != 1)[0]+1)
+    index_ranges_n = [(rng[0], rng[-1]) for rng in ranges]
+    
+
+    index_ranges_p.extend(index_ranges_n)
+    
+    return index_ranges_p
+
+
+
+def get_range_list(ta_flag,_min=0):
+    
+    # starts and ends of non-turn-around parts
+    ones = np.where(ta_flag==False)[0]
+    diff = np.diff(ones)
+    
+    starts = np.insert(ones[np.where(diff != 1)[0]], 0, ones[0])
+    ends = np.append(ones[np.where(diff != 1)[0] - 1], ones[-1])
+    indices = list(zip(starts, ends))
+    indices = [(start, end) for start, end in indices if (end - start + 1 >= _min)]
+
+    return indices
+
+
+def subscan_polyfilter_tamasked(aman, degree, signal='signal', default_flag=False, wrap_flag=True) : 
+    
```

Comment:
We often use a flags=None as default. and 
```Python
if flag is None:
    flag = aman.flags.turnarounds
```
With this configuration, users can use there own flags to mask.

Reply:
So I prefer to merge `subscan_polyfilter_tamasked` and `subscan_polyfilter`.
And masking for glitch should be available for this function.